src/platform/ESP32/nimble/BLEManagerImpl.cpp

```diff
@@ -426,9 +426,37 @@ bool BLEManagerImpl::SendReadResponse(BLE_CONNECTION_OBJECT conId, BLE_READ_REQU
 
 void BLEManagerImpl::NotifyChipConnectionClosed(BLE_CONNECTION_OBJECT conId) {}
 
+CHIP_ERROR BLEManagerImpl::MapBLEError(int bleErr)
```

Comment:
There are 2 types of APIs for NimBLE: Espressif added nimble APIs that return esp_err_t and upstream nimble APIs that return int. You seem to have combined their handling in a single API. See below `BLE_HS_EMSGSIZE` is from nimble and   `ESP_OK` and `ESP_ERR_INVALID_ARG` are from esp-nimble. For now, this should be fine